internal/http/services/owncloud/ocdav/ocdav_blackbox_test.go

```diff
@@ -120,13 +134,383 @@ var _ = Describe("ocdav", func() {
 			},
 		}, nil)
 	})
+	AfterEach(func() {
+		dataSvr.Close()
+	})
 
 	Describe("NewHandler", func() {
 		It("returns a handler", func() {
 			Expect(handler).ToNot(BeNil())
 		})
 	})
 
+	// TODO for every endpoint test the different WebDAV Methods
+
+	// basic metadata
+	// PROPFIND
+	// MKCOL
+	// DELETE
+
+	// basic data
+	// PUT
+	// GET
+	// HEAD
+
+	// move & copy
+	// MOVE
+	// COPY
+
+	// additional methods
+	// PROPPATCH
+	// LOCK
+	// UNLOCK
+	// REPORT
+	// POST (Tus)
+	// OPTIONS?
+
+	Context("at the very legacy /webdav endpoint", func() {
+
+		BeforeEach(func() {
+			// set the webdav endpoint to test
+			basePath = "/webdav"
+
+			// path based requests at the /webdav endpoint first look up the storage space
+			client.On("ListStorageSpaces", mock.Anything, mock.MatchedBy(func(req *cs3storageprovider.ListStorageSpacesRequest) bool {
+				p := string(req.Opaque.Map["path"].Value)
+				return p == "/" || strings.HasPrefix(p, "/users")
+			})).Return(&cs3storageprovider.ListStorageSpacesResponse{
+				Status:        status.NewOK(ctx),
+				StorageSpaces: []*cs3storageprovider.StorageSpace{userspace},
+			}, nil)
+		})
+
+		Describe("PROPFIND", func() {
+
+			var rr *httptest.ResponseRecorder
+			var req *http.Request
+			var err error
+
+			BeforeEach(func() {
+				// setup the request
+				rr = httptest.NewRecorder()
+				req, err = http.NewRequest("PROPFIND", basePath, strings.NewReader(""))
+				Expect(err).ToNot(HaveOccurred())
+				req = req.WithContext(ctx)
+
+			})
+			When("the gateway returns a file list", func() {
+				It("returns a multistatus with the file info", func() {
+
+					client.On("Stat", mock.Anything, mock.MatchedBy(func(req *cs3storageprovider.StatRequest) bool {
```

Comment:
This mock is done quite often. Might be worth to add a helper function for that.

Reply:
introduced, but could  not be used as often

Reply:
added another mockPathStat for the other cases